# CASE STUDY 3

The given case study is to analyze the data which describes about an instituion that need to understand about the math score, reading score, writing score of their children by finding the answer for particular questions. 

In [1]:
#importing modules
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

Read the dataset to python

In [2]:
#reading the data from excel
data=pd.read_csv("StudentsPerformance.csv")

In [3]:
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [6]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [5]:
#finding columns
data.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

In [6]:
#identifying the tail part
data.tail()

gender race/ethnicity parental level of education         lunch  \
995   male        group E                some college      standard   
996   male        group A                some college      standard   
997   male        group D                some college      standard   
998   male        group E          associate's degree  free/reduced   
999   male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100

In [7]:
#finding the shape
data.shape

(1000, 8)

In [8]:
#describing the dataset
data.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000

In [9]:
#finding the amount of values in each case
data.nunique()

gender                          2
race/ethnicity                  5
parental level of education     6
lunch                           2
test preparation course         2
math score                     85
reading score                  72
writing score                  77
dtype: int64

# 1.Find out how many males and females participated in the test!?

In [10]:
data['gender'].unique()

array(['female', 'male'], dtype=object)

In [11]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

Insight:

Number of female=518

Number of male=482

Majority of participants belongs to female category.

# 2.What do you think about'parental level of education'!?

In [12]:

data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

INSIGHT:

1)Majority of the parents had done their colleges.

2)Minorities persuaded their masters. 

3)So that most are able to help their children during their studies,no need of an extra tution.

# 3.Find average based on math,reading,writing score!?

In [5]:
data.groupby("gender")[['math score','reading score','writing score']].mean().round(1)

math score  reading score  writing score
gender                                          
female        63.4           72.6           72.5
male          68.8           65.5           63.4

Insight:

The consistent average value in case of math,reading,writing score is maintainted by male,in case of gender distribution.

In [6]:
data.groupby("test preparation course")[['math score','reading score','writing score']].mean().round(1)

math score  reading score  writing score
test preparation course                                          
completed                      70.0           74.2           74.7
none                           63.8           66.4           64.5

Insight:

The average value in case of test preparation course is maintainted by none category.Because considering math,reading and writing all three showing only a minute variation and also having equal intervals in between three.

In [1]:
import statistics

In [11]:
#average for math score
data1=data['math score'].unique()

In [12]:
data1

array([  0,   8,  18,  11,  22,  23,  24,  26,  19,  29,  32,  33,  34,
        25,  35,  36,  37,  38,  39,  40,  41,  42,  43,  31,  44,  13,
        45,  46,  47,  48,  49,  50,  51,  52,  53,  54,  55,  56,  57,
        58,  59,  60,  61,  62,  63,  64,  65,  66,  67,  68,  69,  70,
        71,  72,  73,  74,  75,  76,  77,  78,  79,  80,  81,  82,  83,
        84,  85,  86,  87,  88,  89,  90,  91,  92,  93,  94,  95,  96,
        97,  98,  99, 100,  27,  28,  30], dtype=int64)

In [13]:
print(statistics.mean(data1))

57


In [17]:
#average for writing score
data2=data['writing score'].unique()

In [18]:
data2

array([ 10,  23,  28,  32,  33,  36,  27,  38,  30,  44,  47,  43,  41,
        39,  54,  46,  60,  56,  50,  45,  57,  51,  64,  48,  40,  69,
        58,  53,  61,  62,  55,  52,  70,  59,  66,  67,  65,  73,  63,
        79,  72,  68,  76,  74,  80,  78,  77,  75,  87,  81,  71,  85,
        84,  82,  83,  86,  88,  89,  90,  94,  95,  91,  98,  93,  96,
        92, 100,  97,  99,  19,  15,  22,  34,  49,  37,  42,  35],
      dtype=int64)

In [19]:
print(statistics.mean(data2))

61


In [20]:
#average for reading score
data3=data['reading score'].unique()

In [21]:
data3

array([ 17,  24,  32,  38,  39,  44,  31,  34,  29,  40,  41,  51,  48,
        42,  61,  53,  55,  57,  46,  45,  43,  60,  49,  64,  52,  58,
        59,  65,  62,  66,  63,  73,  56,  54,  67,  50,  72,  76,  70,
        71,  69,  79,  68,  78,  77,  74,  75,  85,  86,  80,  82,  81,
        84,  83,  89,  90,  87,  92,  88,  94,  97,  91,  93,  95, 100,
        99,  96,  23,  26,  28,  47,  37], dtype=int64)

In [22]:
print(statistics.mean(data3))

62


# 4.Finding the scoring variation!?

* Scoring variation in case of gender:

In [10]:
#average of math,reading,writing score in case of gender
data.groupby("gender")[['math score','reading score','writing score']].mean().round(1)

math score  reading score  writing score
gender                                          
female        63.4           72.6           72.5
male          68.8           65.5           63.4

In [24]:
#minimum in case of gender
data.groupby("gender")[['math score','reading score','writing score']].min()

math score  reading score  writing score
gender                                          
female           0             17             10
male            27             23             15

In [25]:
#maximum in case of gender
data.groupby("gender")[['math score','reading score','writing score']].max()

math score  reading score  writing score
gender                                          
female         100            100            100
male           100            100            100

Insight:

1.Analyzing the data for average values of math,reading,writing score the consistent score is maintainted by male.

2.Analyzing the data in case of minimum values for math,reading,writing the minimum values are gained by females, ie the better ones are males.

3.Analyzing the data in case of maximum values for math,reading,writing the maximum values are 100 in case of both female and male.

 * Scoring variation in case of 'Test preparation course'

In [27]:
#average value
data.groupby("test preparation course")[['math score','reading score','writing score']].mean().round(1)

math score  reading score  writing score
test preparation course                                          
completed                      70.0           74.2           74.7
none                           63.8           66.4           64.5

In [32]:
#minimum value
data.groupby("test preparation course")[['math score','reading score','writing score']].median()

math score  reading score  writing score
test preparation course                                          
completed                        70             75             76
none                             64             67             65

In [29]:
#maximum value
data.groupby("test preparation course")[['math score','reading score','writing score']].max()

math score  reading score  writing score
test preparation course                                          
completed                       100            100            100
none                            100            100            100

Insight:

1.After analyzing the data the one who completed test preparation course has less variation compared to none.


# 5.Find the top 25 to give a bonus in case of math score.

In [32]:
dataset=data.sort_values('math score',ascending=False)

In [33]:
dataset

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
2    female        group B            some high school  free/reduced   
51   female        group C            some high school  free/reduced   
3    female        group B                some college      standard   
1    female        group B                 high school  free/reduced   
0    female        group C            some high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
2                      none          18             32             28  
51                     none          13             41             51  
3                      none          11             38             32  
1                      none           8             24             23  
0                      none           0             17             10  

[1000 rows x 8 columns]

In [34]:
dataset[:25]

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
995    male        group E                some college      standard   
513  female        group E           bachelor's degree      standard   
514  female        group E                 high school      standard   
994    male        group C          associate's degree      standard   
993    male        group C                some college      standard   
512  female        group D                some college      standard   
992    male        group E          associate's degree      standard   
991    male        group E                some college      standard   
989    male        group A          associate's degree      standard   
510  female        group B           bachelor's degree      standard   
511  female        group D            some high school      standard   
990    male        group C          associate's degree      standard   
509  female        group C          associate's degree      standard   
508  female        group C           bachelor's degree      standard   
988    male        group C           bachelor's degree      standard   
507  female        group E          associate's degree      standard   
987    male        group D             master's degree      standard   
505  female        group B          associate's degree      standard   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
997               completed         100             97             99  
998               completed         100            100             93  
995               completed          99             87             81  
513               completed          99            100            100  
514                    none          99             93             90  
994               completed          98             87             90  
993               completed          98             86             90  
512                    none          98            100             99  
992               completed          97             82             88  
991                    none          97             87             82  
989               completed          97             92             86  
510                    none          97             97             96  
511               completed          97            100            100  
990                    none          97             93             91  
509               completed          96             96             99  
508               completed          96            100            100  
988               completed          96             90             92  
507               completed          95             89             92  
987                    none          95             81             84  
505               completed          94             87             92

Insight:

1.Most of them scored between 90-95,majority belongs to that category.

2.In case of gender its give a combination of both male(13) and female(12).

3.In case of test preparation course most of them belongs to completed category(16 completed).